common/elasticsearch/client/v6/client.go

```diff
@@ -0,0 +1,190 @@
+func (c *ElasticV6) PutMapping(ctx context.Context, index, body string) error {
+	_, err := c.client.PutMapping().Index(index).Type("_doc").BodyString(body).Do(ctx)
+	return err
+}
+
+func (c *ElasticV6) CreateIndex(ctx context.Context, index string) error {
+	_, err := c.client.CreateIndex(index).Do(ctx)
+	return err
+}
+
+func (c *ElasticV6) Count(ctx context.Context, index, query string) (int64, error) {
+	return c.client.Count(index).BodyString(query).Do(ctx)
+}
+
+func (c *ElasticV6) ClearScroll(ctx context.Context, scrollID string) error {
+	return elastic.NewScrollService(c.client).ScrollId(scrollID).Clear(ctx)
+}
+func (c *ElasticV6) Scroll(ctx context.Context, index, body, scrollID string) (*client.Response, error) {
+	scrollService := elastic.NewScrollService(c.client)
+	var esResult *elastic.SearchResult
+	var err error
+
+	// we are not returning error immediately here, as result + error combination is possible
+	if len(scrollID) == 0 {
+		esResult, err = scrollService.Index(index).Body(body).Do(ctx)
+	} else {
+		esResult, err = scrollService.ScrollId(scrollID).Do(ctx)
+	}
+
+	if esResult == nil {
+		return nil, err
+	}
+
+	var hits []*client.SearchHit
+	if esResult.Hits != nil {
+		for _, h := range esResult.Hits.Hits {
+			if h.Source != nil {
+				hits = append(hits, &client.SearchHit{Source: *h.Source})
+			}
+		}
+	}
+
+	result := &client.Response{
+		TookInMillis: esResult.TookInMillis,
+		TotalHits:    esResult.TotalHits(),
+		Hits:         &client.SearchHits{Hits: hits},
+		ScrollID:     esResult.ScrollId,
+	}
+
+	if len(esResult.Aggregations) > 0 {
+		result.Aggregations = make(map[string]json.RawMessage, len(esResult.Aggregations))
+		for key, agg := range esResult.Aggregations {
+			if agg != nil {
+				result.Aggregations[key] = *agg
+			}
+		}
+	}
+
+	return result, err
+}
+
+func (c *ElasticV6) Search(ctx context.Context, index, body string) (*client.Response, error) {
+	esResult, err := c.client.Search(index).Source(body).Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if esResult.Error != nil {
+		return nil, types.InternalServiceError{
+			Message: fmt.Sprintf("ElasticSearch Error: %#v", esResult.Error),
+		}
+	} else if esResult.TimedOut {
+		return nil, types.InternalServiceError{
+			Message: fmt.Sprintf("ElasticSearch Error: Request timed out: %v ms", esResult.TookInMillis),
+		}
+	}
+
+	var sort []interface{}
+	var hits []*client.SearchHit
+
+	if esResult != nil && esResult.Hits != nil {
+		for _, h := range esResult.Hits.Hits {
+			hits = append(hits, &client.SearchHit{Source: *h.Source})
+			sort = h.Sort
+		}
+	}
+
+	result := &client.Response{
+		TookInMillis: esResult.TookInMillis,
+		TotalHits:    esResult.TotalHits(),
+		Hits:         &client.SearchHits{Hits: hits},
+		Sort:         sort,
+	}
+
+	if len(esResult.Aggregations) > 0 {
+		result.Aggregations = make(map[string]json.RawMessage, len(esResult.Aggregations))
+		for key, agg := range esResult.Aggregations {
+			result.Aggregations[key] = *agg
```

Comment:
the `nil` check is missing in `Search` func

Reply:
nil check is added